File: pandoc/.local/share/pandoc/lua-filters/center.lua
function Header(h)
    -- Surround headers with centering raw LaTeX
    -- if they have a center{,ed,ing} property set to "t".
    if (h.attributes.centering == "t" or
        h.attributes.centered == "t" or
        h.attributes.center == "t") then
        table.insert(h.content, 1, pandoc.RawInline('latex', '\\centering{'))
        table.insert(h.content, pandoc.RawInline('latex', '}'))
        return h
    end
end

File: pandoc/.local/share/pandoc/clj-filters/src/filter/latex/center.clj
(ns filter.latex.center
  (:require [lib.pandoc :as pandoc :refer [deffilter]]))

(defn- center? [el]
  (let [[_ classes attributes] (pandoc/attrs el)
        keys (map first attributes)
        commands #{"center" "centered" "centering"}]
    (or
     (some commands classes)
     (some commands keys))))

(defn- centering-latex
  "Insert raw LaTeX centering command around the input element."
  [el]
  (vec
   (concat
    [(pandoc/raw-inline "latex" "\\centering{")]
    el
    [(pandoc/raw-inline "latex" "}")])))

(deffilter pandoc/header? [el]
  {:if (center? el)}
  (pandoc/update-inlines el centering-latex))
